### PR TITLE
models

### PR DESCRIPTION
**Pull request by Godhanded**

### CODE REVIEW

**Comment by Godhanded** [comment]
@pauline-banye @SanCrystal please review

File: src/schema/questions_model.py [hunk 1]
```diff
@@ -11,14 +11,26 @@ class Level(str, Enum):
...
+    d:str
+
+class test_question(BaseModel):
+    id:int
+    question:str
+    options: option
+    answer: str
 
-class Question(BaseModel):
+class test(BaseModel):
```
Comment (SanCrystal) [inline src/schema/questions_model.py]: Capitalize your class name

File: src/schema/questions_model.py [hunk 2]
```diff
@@ -11,14 +11,26 @@ class Level(str, Enum):
     INTERMEDIATE = "INTERMEDIATE"
     ADVANCED = "ADVANCED"
 
+class option(BaseModel):
+    a:str
+    b:str
+    c:str
+    d:str
+
+class test_question(BaseModel):
```
Comment (SanCrystal) [inline src/schema/questions_model.py]: Pascal case for class names

File: src/schema/questions_model.py [hunk 3]
```diff
@@ -11,14 +11,26 @@ class Level(str, Enum):
     INTERMEDIATE = "INTERMEDIATE"
     ADVANCED = "ADVANCED"
 
+class option(BaseModel):
```
Comment (SanCrystal) [inline src/schema/questions_model.py]: Capitalize your class name

File: src/schema/questions_model.py (hunk 2 above)
Comment (SanCrystal) [inline src/schema/questions_model.py]: Add a doc string

File: src/schema/questions_model.py (hunk 3 above)
Comment (SanCrystal) [inline src/schema/questions_model.py]: Add a doc string

File: src/schema/questions_model.py (hunk 1 above)
Comment (SanCrystal) [inline src/schema/questions_model.py]: Add a doc string

**Comment by Godhanded** [comment]
created a cluster and collection on mongo d, no errors, shows success code but the data is not being recorded on mongo. review @pauline-banye @SanCrystal something missing?

File: src/controller/users_controls.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from fastapi.encoders import jsonable_encoder
+from src.utils.functions import model
+from ..schema.users_model import user
+from ..utils.auth import hash
+    
+async def register(User):
+    try:
+        User=user()
+        mail= User.email
+        check= await model.findone(mail,"questions")
```
Comment (SanCrystal) [inline src/controller/users_controls.py]: Why are we looking for user/searching for user with a question?

File: src/controller/users_controls.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        if check is not None:
+            return {"error: ": "user already exists"}
+        User.password = hash(User.password)
+        foo= await model.create(jasonable_encoder(User), "questions")
```
Comment (SanCrystal) [inline src/controller/users_controls.py]: Jasonable looks incorrect here, did you test?

Comment (SanCrystal) [inline src/controller/users_controls.py]: Jsonable_encoder

File: src/controller/users_controls.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from fastapi.encoders import jsonable_encoder
+from src.utils.functions import model
+from ..schema.users_model import user
+from ..utils.auth import hash
+    
+async def register(User):
```
Comment (SanCrystal) [inline src/controller/users_controls.py]: Add a doc to the function

**Review by SanCrystal — COMMENTED** [review]
Awesome work so far

File: src/config/settings.py [hunk 7]
```diff
@@ -13,7 +13,7 @@ class Settings(BaseSettings):
 
     PROJECT_NAME = "Verificate"
     MONGO_SERVER = os.getenv("MONGO_SERVER")
-    SECRET_KEY = os.getenv("SECRET_KEY")
+    #SECRET_KEY = os.getenv("SECRET_KEY")
```
Comment (paulinebanye) [inline src/config/settings.py]: why is the secret key commented out?

File: src/controller/users_controls.py (hunk 6 above)
Comment (paulinebanye) [reply src/controller/users_controls.py]: pylint errors would crop up without docstrings

File: src/controller/users_controls.py (hunk 5 above)
Comment (paulinebanye) [reply src/controller/users_controls.py]:
```suggestion
        foo= await model.create(jsonable_encoder(User), "questions")
```
why are we using foo? Please use a more descriptive name

File: src/schema/questions_model.py (hunk 3 above)
Comment (paulinebanye) [inline src/schema/questions_model.py]:
```suggestion
class Option(BaseModel):
```

File: src/schema/questions_model.py (hunk 2 above)
Comment (paulinebanye) [inline src/schema/questions_model.py]:
```suggestion
class Test_question(BaseModel):
```

File: src/schema/users_model.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+from datetime import datetime, timezone
+
+from pydantic import BaseModel, EmailStr, Field
+from .questions_model import Level
+
+from pydantic import BaseModel
+
+class user(BaseModel):
```
Comment (paulinebanye) [inline src/schema/users_model.py]:
```suggestion
class User(BaseModel):
```

File: src/schema/users_model.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    name: str = Field(min_length=10)
+    password: str = Field(min_length=7)
+    email: EmailStr
+    linked_in:str = Field(min_length=8)
+    skill: list[str]
+    level:Level
+    created_at: str = str(datetime.now(timezone.utc))
+
+
+class verifier(BaseModel):
```
Comment (paulinebanye) [inline src/schema/users_model.py]:
```suggestion
class Verifier(BaseModel):
```

**Review by paulinebanye — COMMENTED** [review]
Welldone! just some changes @Godhanded.
note* for python, all classnames should be in captial letters

File: src/controller/users_controls.py (hunk 4 above)
Comment (Godhanded) [reply src/controller/users_controls.py]: that was the name of the collection was testing with,resolved

File: src/controller/users_controls.py (hunk 5 above)
Comment (Godhanded) [reply src/controller/users_controls.py]: resolved

File: src/config/settings.py (hunk 7 above)
Comment (Godhanded) [reply src/config/settings.py]: wasn't in use at that time

**Comment by Godhanded** [comment]
@SanCrystal @pauline-banye updates, review

File: src/schema/users_model.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+class User(BaseModel):
+    """
+    Model to hold  user data.
+    """
+
+    name: str = Field(min_length=10)
+    password: str = Field(min_length=7)
+    email: EmailStr
+    linked_in: str = Field(min_length=8)
```
Comment (paulinebanye) [inline src/schema/users_model.py]: linkedin should be httpurl not str

File: src/schema/users_model.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    skill: list[str]
```
Comment (paulinebanye) [inline src/schema/users_model.py]: if were having a list of strings, this list would be populated from the FE? we'll need an update route to add more skills I think. how about if skills is a standalone model, that can be updated by itself?

File: src/schema/users_model.py [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+from src.schema.questions_model import Level
+
+
+class User(BaseModel):
+    """
+    Model to hold  user data.
+    """
+
+    name: str = Field(min_length=10)
```
Comment (paulinebanye) [inline src/schema/users_model.py]: can we separate these fields? first name and last name? some people may not put a space btw. i'm sure front end can concat their input n generate the gull name

**Review by paulinebanye — COMMENTED** [review]
go through @Godhanded . haven't pulled to run it tho.

File: src/schema/users_model.py (hunk 10 above)
Comment (Godhanded) [reply src/schema/users_model.py]: done

File: src/schema/users_model.py (hunk 12 above)
Comment (Godhanded) [reply src/schema/users_model.py]: done

File: src/schema/users_model.py [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+    """
+    Model to hold  user data.
+    """
+
+    first_name: str = Field(min_length=10)
+    last_name: str = Field(min_length=10)
+    password: str = Field(min_length=7)
+    email: EmailStr
+    linked_in: HttpUrl
+    results: Test_results | None = None
```
Comment (SanCrystal) [inline src/schema/users_model.py:35]: List of results

File: src/schema/users_model.py [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+    INTERMEDIATE: bool = False
+    ADVANCED: bool = False
+
+
+class Test_results(BaseModel):
+    """
+    to hold eaxh categories test results
+    """
+
+    email: EmailStr
```
Comment (SanCrystal) [inline src/schema/users_model.py:21]: Should have a status, eg passed etc

File: src/schema/users_model.py [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    category: lvls
```
Comment (SanCrystal) [inline src/schema/users_model.py:22]: Score

File: src/schema/users_model.py [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+    """
+    true boolean if beginer exam is passed.
+    """
+
+    BEGINNER: bool = False
+    INTERMEDIATE: bool = False
+    ADVANCED: bool = False
+
+
+class Test_results(BaseModel):
```
Comment (SanCrystal) [inline src/schema/users_model.py:16]: Pascal case for class names

File: src/schema/users_model.py [hunk 17]
```diff
@@ -0,0 +1,55 @@
+from datetime import datetime, timezone
+
+from pydantic import BaseModel, EmailStr, Field, HttpUrl
+
+
+class lvls(BaseModel):
```
Comment (SanCrystal) [inline src/schema/users_model.py:6]: Name should convey a meaning an explicit one 🫡 also maintain pascal convention for naming classes

**Pull request merged**